services.php

```diff
@@ -26,10 +28,15 @@
 $templateLoader = new Twig_Loader_Filesystem(__DIR__ . "/src/View");
 $twig = new Twig_Environment($templateLoader);
 
+//FormHandlers
+$threadFormHandler = new ThreadFormHandler($entityManager);
+
 //Application Controllers
 $mainController = new MainController($twig, $threadRepository);
+$threadController = new ThreadController($twig, $threadRepository, $threadFormHandler);
 
 //Application
 $application = new Application(['debug' => true]);
 $router = new Router($application);
-$router->routeMainController($mainController);
\ No newline at end of file
+$router->routeMainController($mainController);
+$router ->routePostController($threadController);
```

Comment:
If each of these functions return $this you can do chaining which is a nice pattern:
```
$router
    ->routeMainController($mainController)
    ->routePostController($threadController)
;
```